barretenberg/cpp/src/barretenberg/circuit_checker/ultra_circuit_checker.cpp

```diff
@@ -19,7 +19,7 @@ template <typename Builder> bool UltraCircuitChecker::check(const Builder& build
 {
     // Create a copy of the input circuit and finalize it
     Builder builder{ builder_in };
-    builder.finalize_circuit();
+    builder.finalize_circuit(/*ensure_nonzero=*/false); // No need to add nonzero gates if we're not proving
```

Comment:
hmm is there any harm in it though? if there were a bug in one of the added gates this would help us catch it

Reply:
Was just keeping it how things were, but this is a fantastic point